src/select/select.js

```diff
@@ -736,20 +744,39 @@ class Select extends React.Component<PropsT, SelectStateT> {
 
   filterOptions(excludeOptions: ?ValueT) {
     const filterValue = this.state.inputValue;
-    const options = this.props.options || [];
+    var options = this.props.options || [];
+    // apply filter function
     if (this.props.filterOptions) {
-      return this.props.filterOptions(options, filterValue, excludeOptions, {
+      options = this.props.filterOptions(options, filterValue, excludeOptions, {
         valueKey: this.props.valueKey,
         labelKey: this.props.labelKey,
       });
-    } else {
-      return options;
     }
+    // can user create a new option + there's no exact match already
+    if (
+      filterValue &&
+      this.props.creatable &&
+      options.every(
+        opt =>
+          opt[this.props.labelKey].toLowerCase() !==
+          filterValue.toLowerCase().trim(),
+      )
+    ) {
+      // $FlowFixMe
+      options.push({
+        id: filterValue,
+        [this.props.labelKey]: filterValue,
+        [this.props.valueKey]: filterValue,
```

Comment:
`valueKey` is the `id` by default

Reply:
Hm, if `valueKey = id` then it just gets overridden and everything is fine?

Or we just don't need `id` and should delete the

```js
id: filterValue
```

row?

Reply:
Yes, it still works as expected, just duplicated keys 🤷‍♀️